File: NeetCode/Stack/Crawler_Log/Crawler_Log.cpp
// Crawler_Log.cpp
//
// Compile locally with:
// clang++ Crawler_Log.cpp -std=c++17 -Wall -Wextra -O2 -o run && ./run
//
// NOTE:
// - Remove main() before submitting to LeetCode.
// - Keep only the class Solution definition.
//

#include <algorithm>
#include <array>
#include <cassert>
#include <cctype>
#include <climits>
#include <cmath>
#include <deque>
#include <functional>
#include <iostream>
#include <map>
#include <numeric>
#include <queue>
#include <set>
#include <stack>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>

using namespace std;

class Solution
{
public:
    int minOperations(vector<string> &logs)
    {
        int stk = 0;
        for (string &log : logs)
        {
            if (log == "../")
            {
                if (stk)
                {
                    stk--;
                }
                else
                {
                    continue;
                }
            }
            else if (log == "./")
            {
                continue;
            }
            else
            {
                stk++;
            }
        }
        return stk;
    }
};

int main()
{
    cout << "Running Crawler_Log.cpp..." << endl;

    Solution sol;

    // TODO:
    // Add local test calls here

    return 0;
}

File: NeetCode/Stack/Crawler_Log/crawler__log.py
from typing import Optional, List
import heapq
from collections import defaultdict


class Solution:
    def minOperations(self, logs: List[str]) -> int:
        stk = 0

        for log in logs:
            if log == "../":
                if stk > 0:
                    stk -= 1
                else:
                    continue
            elif log == "./":
                continue
            else:
                stk += 1

        return stk


if __name__ == "__main__":
    sol = Solution()
    print("Running Solution...")
